reflex/app.py

```diff
@@ -654,6 +703,9 @@ def compile_(self):
         if not self._should_compile():
             return
 
+        self._setup_overlay_component()
+        self._setup_dynamic_routes()
```

Comment:
I think this needs to happen _above_ the `if not self._should_compile()` line, because the backend needs to setup the dynamic route vars, even if we're not compiling the frontend.

Reply:
But we only need to update them if something changed or am I missing something? I tested it with the integration tests and one of my apps which uses dynamic routes.

Reply:
did you test it with an export and `reflex run --backend-only`?

When running in backend only mode, the compile step is skipped.

Reply:
You got me, I did not test a prod deployment. We should think about adding a test for this.